python/cuspatial/cuspatial/tests/binpreds/test_contains_properly.py

```diff
@@ -349,3 +350,210 @@ def test_max_polygons_max_multipoints(multipoint_generator, polygon_generator):
     got = lhs.contains_properly(rhs).values_host
     expected = gpdlhs.contains(gpdrhs).values
     assert (got == expected).all()
+
+
+@pytest.mark.parametrize(
+    "object",
+    [
+        Polygon([[0, 0], [1, 1], [1, 0], [0, 0]]),
+        MultiPolygon(
+            [
+                Polygon([[0, 0], [1, 1], [1, 0], [0, 0]]),
+                Polygon([[0, 0], [1, 1], [1, 0], [0, 0]]),
+            ]
+        ),
+    ],
+)
+def test_self_contains(object):
+    gpdobject = gpd.GeoSeries(object)
+    object = cuspatial.from_geopandas(gpdobject)
+    got = object.contains_properly(object).values_host
+    expected = gpdobject.contains(gpdobject).values
+    assert not got
+    assert expected
```

Comment:
We must use `np.testing.assert_arrays_equal` to test here (or other explicit tests). Because Python will recognize all non Falsy object as True. And that means `assert expected` will always be true. And `assert not got` might still be true even if you return `None` for `contains_properly`.

```suggestion
    np.testing.assert_arrays_equal(got, np.array([False]))
    np.testing.assert_arrays_equal(expected, np.array([True]))
```